metabase/metabase
# Bulk filter modal incorrectly suggests that you can filter by values in a JSON column

Fixes https://github.com/metabase/metabase/issues/38111

Previously the type system worked in a way that structured field types (JSON, XML) were considered strings. This made the FE show filter widgets for these types but all queries would fail. As only `is-null` and `not-null` are allowed, this PR makes a change to no longer treat `:type/Structured` as `:type/Text`.

How to verify:
- Postgres with a json field
- Make sure that the UI only allows to use `is-null` and `not-null` (which are called `Is empty` and `Not empty` in the UI).

<img width="994" alt="Screenshot 2024-07-17 at 15 28 29" src="https://github.com/user-attachments/assets/8be32a65-464f-4725-87df-fd5beb7f8f5f">
